cost/forecasting/commitment_forecast/commitment_forecast.pt

```diff
@@ -0,0 +1,394 @@
+
+  var result = []
+  _.each(total_cost, function(tc){
+    result.push({
+      "commitment_period": period_length,
+      "period_completed": period_completed_to_date,
+      "actual_spend": tc.total,
+      "forecasted_spend": (tc.total * (1/period_completed_to_date)) - tc.total, 
+      "forecasted_total": tc.total * (1/period_completed_to_date) 
+      "commitment_target": commitment_target 
+      "name": tc.name,
+      "vendor": tc.vendor
+    }) 
+  })
+  EOS
+end
+
+#CREATE AND FORMAT CHART DATA
+datasource "ds_chart_data" do
+  run_script $js_get_chart_data, $ds_actual_forecasted_spend, $ds_currency_reference, $ds_currency_code, $param_commitment_target
+end
+
+script "js_get_chart_data", type: "javascript" do
+  parameters "spend_data", "curr_ref", "curr_code", "commitment_target"
+  result "report"
+  code <<-EOS
+  var spend_diff = spend_data[0].forecasted_total - commitment_target
+  var curr = "$", separator = ",", target_name = "", over_commitment_data = "", over_commitment_label = "", over_commitment_amount_value = "", chart_data_values = ""
+  
+  if (curr_code.value !== undefined){
+    curr = curr_ref[curr_code.value].symbol,
+    separator = curr_ref[curr_code.value].t_separator
+  }
+
+  function formatNumber(number, separator){
+    var values = number.toString().split(".");
+    var result = ''
+
+    while (values[0].length > 3){
+      var chunk = values[0].substr(-3)
+      values[0] = values[0].substr(0, values[0].length - 3)
+      result = separator + chunk + result
+    }
+    if (values[0].length > 0){
+      result = values[0] + result
+    }
+    if(values[1]==undefined){
+      return result
+    }
+    return result + "." + values[1]
+  }
+
+  if(spend_diff < 0 ){
+    target_name = "Not Met"
+  } else {
+    target_name = "Met"
+  }
+
+  if(target_name == "Met"){
+    over_commitment_data = 0+","+(spend_diff.toFixed(2))
+    over_commitment_label = "Forecasted+Amount+Over+Commitment+Target"
+    over_commitment_amount_value = spend_diff.toFixed(2)
+    over_commitment_color = "4CBA54"
+    chart_data_values = (curr+formatNumber(spend_data[0].actual_spend.toFixed(2),separator)) +'||' + (curr+formatNumber(spend_data[0].forecasted_spend.toFixed(2),separator)) +'|||'+ (curr+formatNumber(commitment_target,separator)) +'||'+ (curr+formatNumber(over_commitment_amount_value,separator))
+  } else {
+    over_commitment_data = (spend_diff.toFixed(2)*-1)+","+0
+    over_commitment_label = "Forecasted+Amount+Under+Commitment+Target"
+    over_commitment_amount_value = spend_diff.toFixed(2)*(-1)
+    over_commitment_color = "F04B46"
+    chart_data_values = (curr+formatNumber(spend_data[0].actual_spend.toFixed(2),separator)) +'||' + (curr+formatNumber(spend_data[0].forecasted_spend.toFixed(2),separator)) +'|||'+ (curr+formatNumber(commitment_target,separator)) +'|'+ (curr+formatNumber(over_commitment_amount_value,separator) + '|')
+  }
+
+  report = {
+    chart_type: "cht=bhs",
```

Comment:
Can you go ahead and URIEncode this to future proof it?

Reply:
Done.